rosbag2_transport/src/rosbag2_transport/qos.hpp

```diff
@@ -0,0 +1,66 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ROSBAG2_TRANSPORT__QOS_HPP_
+#define ROSBAG2_TRANSPORT__QOS_HPP_
+
+#include "rclcpp/qos.hpp"
+
+#ifdef _WIN32
+// This is necessary because of a bug in yaml-cpp's cmake
+#define YAML_CPP_DLL
+// This is necessary because yaml-cpp does not always use dllimport/dllexport consistently
+# pragma warning(push)
+# pragma warning(disable:4251)
+# pragma warning(disable:4275)
+#endif
+#include "yaml-cpp/yaml.h"
+#ifdef _WIN32
+# pragma warning(pop)
+#endif
+
+
+namespace rosbag2_transport
+{
+/// Simple wrapper around rclcpp::QoS to provide a default constructor for YAML deserialization.
+class Rosbag2QoS : public rclcpp::QoS
+{
+public:
+  Rosbag2QoS()
+  : rclcpp::QoS(10) {}
```

Comment:
Why 10? We should probably document the rationale of this constant, to make sure this knowledge is not lost. Could you also use something like QoS(10 /* arg_name */) to help the reader? I have no idea what this means here.

Do we always overwrite this 10 when the YAML obj is deserialized? If yes, maybe that's also something to explain.

Reply:
Yes, we'll always overwrite it when we deserialize, it's an arbitrary number (that is used pretty widely as a de facto default), but it's part of the rclcpp api that users _must_ specify a depth, hence this wrapper for the yaml library that tries to use an empty constructor. I'll add comments to that effect